pkg/directory/setup.go

```diff
@@ -43,6 +43,7 @@ func Setup(parent *mux.Router, db *mongo.Database) error {
 	farmsAuthenticated.HandleFunc("/{node_id}", mw.AsHandlerFunc(nodeAPI.Requires("node_id", farmAPI.deleteNodeFromFarm))).Methods("DELETE").Name("farm-node-delete-v1")
 	farmsAuthenticated.HandleFunc("/deals", mw.AsHandlerFunc(farmAPI.createOrUpdateFarmCustomPrice)).Methods("POST", "PUT").Name("farm-update-prices-v1")
 	farmsAuthenticated.HandleFunc("/deals/{threebot_id}", mw.AsHandlerFunc(farmAPI.deleteFarmCustomPrice)).Methods("DELETE").Name("farm-delete-prices-v1")
+	farmsAuthenticated.HandleFunc("/dedicate-node", mw.AsHandlerFunc(farmAPI.setNodeDedicated)).Methods("POST").Name("farm-set-node-dedicated")
```

Comment:
I think a better url should be `/{node_id}/dedicated/` then a struct with the user id